### ISSUE TITLE
intersphinx's default link is not stable by each build when intersphinx_mapping have plural items.

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1029]
- This issue comes up by Python-3.3

At Python-3.3, test_missing_reference test sometime fail:

```
======================================================================
FAIL: test_intersphinx.test_missing_reference
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/bpbook/sphinx-fix-testing-fork/.tox/py33/lib/python3.3/site-packa>
    self.test(*self.arg)
  File "/home/bpbook/sphinx-fix-testing-fork/.tox/py33/lib/python3.3/site-packa>
    return func(*arg, **kw)
  File "/home/bpbook/sphinx-fix-testing-fork/.tox/py33/tests/util.py", line 187>
    func(app, *args2, **kwargs2)
  File "/home/bpbook/sphinx-fix-testing-fork/.tox/py33/tests/util.py", line 214>
    func(tempdir, *args, **kwds)
  File "/home/bpbook/sphinx-fix-testing-fork/.tox/py33/tests/test_intersphinx.p>
    ('foo', '2.0', 'http://docs.python.org/foo.html#module-module2', '-')
AssertionError
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1029
- Originally reported by: Takayuki Shimizukawa
- Originally created at: 2012-10-30T04:34:25.641

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-10-30 03:48:31+00:00_

This caused by `load_mappings` function in `ext/intersphinx.py`. The function create cache dictionary named `env.intersphinx_inventory` that resolve non-named missing references.

But, sometime (by dict.iteritem() return value's ordering) `env.intersphinx_inventory` values are overwritten by named mapping values.

This behavior was implemented at [ext/intersphinx.py L192-196](https://github.com/sphinx-doc/sphinx/blob/0c02841412f8c0efe871853f192252e58e23680d/sphinx/ext/intersphinx.py#L192)

Patch is here: [shimizukawa fork 7685daff91f2](https://bitbucket.org/shimizukawa/sphinx-fix-testing-fork/changeset/7685daff91f2bbdefc1ae5470d14ca93273da345)

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-11-01 16:54:57+00:00_

fix #1008 #1029 : intersphinx_mapping values are not stable if mapping have plural key/value set

→ <<cset d856e89a79768f572e627ca927d14ff51f6892cb>>
